src/Schemas.ts

```diff
@@ -510,6 +510,36 @@ export function union<TL, TR, SR>(left: Schema<TL, TL>, right: Schema<TR, SR>):
     );
 }
 
+type UnionDomainsReprs<Schemas extends unknown[], Acc = never> =
+    Schemas extends [infer H, ...(infer T)]
+        ? H extends Schema<infer D, infer D>
+            ? UnionDomainsReprs<T, Acc | D>
+            : never
+        : Acc;
+
+/**
+ * Construct a schema for a type union given any number of schemas. The schemas
+ * must be trivially encodable ('Schema<T, T>'). Like 'union' and 'unionOf',
+ * this is left-biased; the first (leftmost) schema will apply in the case
+ * where more than one schema would validate a value.
+ */
+export function multiUnion<
+    R extends NonEmptyArray<Schema<any, any>>
```

Comment:
Can you add a note here about the `any`s? There's a detailed note in `recordOf`, you can just point there like is done in `tupleOf`